### PR TITLE
feat: Create a conversation with members from federated backends (SQCORE-773)

### PR DESCRIPTION
**Pull request by bennycode**

### CODE REVIEW

File: src/script/conversation/ConversationFilter.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+export function isGroupWithFederatedUser(conversationEntity: Conversation, userEntity: User): boolean {
+  if (!conversationEntity.is1to1()) {
+    const [userId] = conversationEntity.participatingQualifiedUserIds();
+    return userEntity.id === userId?.id;
+  }
+  return false;
+}
+
+export function isInTeam(conversationEntity: Conversation, userEntity: User): boolean {
```
Comment (Yserz) [inline src/script/conversation/ConversationFilter.ts:47]: looks like a duplicate of `isTeamConversation` https://github.com/wireapp/wire-webapp/pull/11279/files#diff-2ee7e1e0e8d5c3bf23bd4abe48ebba8253759a6006abfebaa1f56d5b307c0757R23

File: src/script/conversation/ConversationMapper.ts [hunk 2]
```diff
@@ -316,8 +324,21 @@ export class ConversationMapper {
         const noOthers = !mergedConversation.others || !mergedConversation.others.length;
         if (isGroup || noOthers) {
           mergedConversation.others = othersStates
-            .filter(otherState => (otherState.status as number) === (ConversationStatus.CURRENT_MEMBER as number))
+            .filter(otherState => {
+              return (
+                (otherState.status as number) === (ConversationStatus.CURRENT_MEMBER as number) &&
```
Comment (Yserz) [inline src/script/conversation/ConversationMapper.ts:329]: `!isRemovedFromConversation(otherState)`

File: src/script/conversation/ConversationFilter.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*
```
Comment (Yserz) [inline src/script/conversation/ConversationFilter.ts:1]: WDYT about putting this into the `UserState` or `ConversationState`?

File: src/script/entity/User.ts [hunk 4]
```diff
@@ -113,14 +113,15 @@ export class User {
     };
   }
 
-  constructor(id: string = '') {
+  constructor(id: string = '', domain: string | null) {
```
Comment (Yserz) [inline src/script/entity/User.ts:116]: Why do we put explicit `null`s instead of optional `domain` here?

Comment (bennycode) [reply src/script/entity/User.ts:116]: We wanted to identify all constructors which potentially need to supply a domain.

File: src/script/conversation/ConversationFilter.ts (hunk 3 above)
Comment (bennycode) [reply src/script/conversation/ConversationFilter.ts:1]: The `ConversationFilter` has only stateless functions so its functions are not a good fit for the `ConversationState` which is stateful.

File: src/script/conversation/ConversationFilter.ts (hunk 1 above)
Comment (bennycode) [reply src/script/conversation/ConversationFilter.ts:47]: The `isTeamConversation` checks if a conversation belongs to a team but `isInTeam` checks if a user is in the same team to which the conversation belongs to.

**Pull request merged**